yellowpages/yellowpages-war/src/main/java/org/silverpeas/components/yellowpages/control/YellowpagesSessionController.java

```diff
@@ -865,13 +865,12 @@ private LocalizationBundle getDomainMultilang() {
   }
 
   public String exportAsCSV() {
-    List<StringBuilder> csvRows = exportAllDataAsCSV();
-
+    List<StringBuilder> csvRows = exportCurrentContactsAsCSV();
     return writeCSVFile(csvRows);
   }
 
-  private List<StringBuilder> exportAllDataAsCSV() {
-    Collection<ContactFatherDetail> contacts = getAllContactDetails(currentTopic.getNodePK());
+  private List<StringBuilder> exportCurrentContactsAsCSV() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 38 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3776)